### PR TITLE
Elastic farm v3

### PR DESCRIPTION
**Pull request by XiaoYhun**

### CODE REVIEW

File: src/pages/Farm/ElasticFarmv2/components/ListView.tsx [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+            return (
+              <Flex
+                alignItems="center"
+                sx={{ gap: '2px' }}
+                color={range.isRemoved ? theme.warning : theme.subText}
+                fontSize={12}
+                fontWeight="500"
+              >
+                {convertTickToPrice(farm.token0, farm.token1, range.tickLower, farm.pool.fee)}
+                <svg width="16" height="16" viewBox="0 0 16 16" fill="none" display="block">
```
Comment (hungdoansy) [inline src/pages/Farm/ElasticFarmv2/components/ListView.tsx:204]: Please check these SVGs as well

File: src/pages/Farm/ElasticFarmv2/components/StakeWithNFTsModal.tsx [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+        setAttemptingTxn(false)
+        setTxHash(txHash || '')
+      })
+      .catch(e => {
+        setAttemptingTxn(false)
+        setErrorMessage(e?.message || JSON.stringify(e))
+      })
+  }, [farm, activeRange, deposit, selectedPosArray])
+
+  const addliquidityElasticPool = `${APP_PATHS.ELASTIC_CREATE_POOL}/${
```
Comment (hungdoansy) [inline src/pages/Farm/ElasticFarmv2/components/StakeWithNFTsModal.tsx:284]:
```suggestion
  const addliquidityElasticPool = `${networkInfo.route}${APP_PATHS.ELASTIC_CREATE_POOL}/${
```

File: src/components/ProAmm/ProAmmPoolInfo.tsx [hunk 3]
```diff
@@ -55,38 +55,14 @@ export default function ProAmmPoolInfo({
...
-    }
-
     return (
-      <MouseoverTooltip width="fit-content" placement="top" text={t`Available for yield farming`}>
-        <Link to={`${APP_PATHS.FARMS}/${networkInfo.route}?tab=elastic&type=active&search=${poolAddress}`}>
-          <FarmingIcon />
-        </Link>
-      </MouseoverTooltip>
+      <Flex sx={{ gap: '4px' }} alignItems="center">
+        {(isFarmActive || isFarmV2Active) && <FarmTag address={poolAddress} />}
```
[the remark this answers is not in the document]
Comment (viet-nv) [reply src/components/ProAmm/ProAmmPoolInfo.tsx:64]: what does this mean @hungdoansy?

**Review by hungdoansy — APPROVED** [review]

**Review by viet-nv — APPROVED** [review]

File: src/pages/AddLiquidityV2/index.tsx [hunk 4]
```diff
@@ -1121,6 +1249,11 @@ export default function AddLiquidity() {
   }
 
   const poolStat = poolDatas?.[poolAddress] || poolDatas?.[poolAddress.toLowerCase()]
+  const poolStatRef = useRef(poolStat)
+  if (poolStat) {
+    poolStatRef.current = poolStat
```
Comment (namgold) [inline src/pages/AddLiquidityV2/index.tsx:1254]: Can I have some reason :D

**Pull request merged**